### PR TITLE
fix(gazelle): _actually_ don't write narrow dependencies.

### PR DESCRIPTION
**Pull request by aryamccarthy**
Sorry about #1681 - I had uploaded an older version. This one contains the correct behavior, in line with what I had written.

The goal is to inherit visibility from the BUILD file's default_visibility (or whatever defaults exist higher in the tree).

I'm happy to discuss the issue further at #1682 if you have apprehensions about this change.

### CODE REVIEW

**Comment by aryamccarthy** [comment]
Note that every failing test right now is for the same reason. They look like this: 

              	py_binary(
              	    name = "main",
              	    srcs = ["main.py"],
            - 	    visibility = ["//:__subpackages__"],
              	    deps = ["@pip//:pandas"],
              	)

This is because the auto-generated visibility lines aren't produced by this PR. That's the intended behavior, but `grep`-ing for the test names didn't turn anything up. Otherwise, I'd update the tests myself.

File: gazelle/python/generate.go [hunk 1]
```diff
@@ -212,7 +212,6 @@ func (py *Python) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	}
 
 	parser := newPython3Parser(args.Config.RepoRoot, args.Rel, cfg.IgnoresDependency)
-	visibility := fmt.Sprintf("//%s:__subpackages__", pythonProjectRoot)
```
Comment (aignas) [inline gazelle/python/generate.go:215]: Actually, this is desirable in usecases where you have a monorepo, where you have many requirements.txt files that you don't want to mix. This change would be a regression in that front.

The `go` plugin for `gazelle` has directives to control the visibility as is documented in https://github.com/bazelbuild/bazel-gazelle, maybe introducing a directive would be useful here as well?

**Review by aignas — CHANGES_REQUESTED** [review]
The tests that are failing are in `gazelle/python/testdata` if I remember correctly.

File: gazelle/python/generate.go (hunk 1 above)
Comment (eaplatanios) [reply gazelle/python/generate.go:215]: With this PR you can still get the old behavior by adding this to your Python root BUILD file:
```
package(default_visibility = ["//" + package_name() + ":__subpackages__"])
```
There's some more context in [this previous PR](https://github.com/bazelbuild/rules_python/issues/1682) that accidentally included the wrong change but the right PR description.

**Comment by aignas** [comment]
Let me know if we can close this since #1784 got merged. i think that should address your use-case.

**Pull request closed without merging**